server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -171,11 +171,13 @@ private void doRecovery(final long recoveryId) {
             final RecoveryTarget recoveryTarget = recoveryRef.target();
             timer = recoveryTarget.state().getTimer();
             cancellableThreads = recoveryTarget.cancellableThreads();
+            final IndexShard indexShard = recoveryTarget.indexShard();
```

Comment:
I think this might need to remain within the try block, as `indexShard()` can throw an exception.

Reply:
Good catch - I should not have moved that